go/internal/services/usagelimiter/redis.go

```
@@ -0,0 +1,380 @@
+func (s *counterService) handleDecrementResult(req UsageRequest, redisKey string, remaining int64) (UsageResponse, error) {
+	if remaining >= 0 {
+		// Success - buffer the credit change for async DB update
+		s.replayBuffer.Buffer(CreditChange{
+			KeyID:  req.KeyId,
+			Amount: req.Cost,
+		})
+
+		metrics.UsagelimiterDecisions.WithLabelValues("redis", "allowed").Inc()
+		metrics.UsagelimiterCreditsProcessed.Add(float64(req.Cost))
+
+		return UsageResponse{Valid: true, Remaining: int32(remaining)}, nil
```

golangci-lint (2.2.2)

[high] G115: integer overflow conversion int64 -> int32

(gosec)